### PR TITLE
Bones from the gibber

### LINKED ISSUE #16273: If you put bonemen in the gibber you get meat.
(WEB REPORT BY: meetmeindorms REMOTE: 172.93.109.202:7777)
# Revision
8984ad596abd59f4e21c488640f4fd58d4feba9d

# Description
Put skellingtons in the gibber and you get meat.
# Steps to Reproduce
1. Crack the ack.
2. Put cracked ack in the gibber.
3. Meat

# What you Expected
I have no clue what I expected.

# What Actually Happened
Meat

### PR DESCRIPTION
**Pull request by Probe1**
**Adds** bones so skellingtons can be gibbed into not-meat.
Resolves https://github.com/vgstation-coders/vgstation13/issues/16273

Thanks @NigglyWiggly for the sprites, @Kurfursten for the code direction, and @DrCelt for something I'm sure.

🆑

* bugfix: Skellingtons stuffed into the gibber now produce bones instead of human meat.  Credit to NigglyWiggly for the sprites.
* rscadd: There are now sheets of bones that can be ground up into bone marrow.

### CODE REVIEW

**Comment by DrSnips** [comment]
I like the idea of bone items. It could be added to limbs as an "organ" in them that you could remove for reasons. Deboned limbs would act like decloned limbs.

**Comment by Probe1** [comment]
I'm going to stop before going all out into bone shivs and bone wind chimes and bone marrow to improve plant growth.  But that did remind me to check and see if limbs can be butchered for meat since I know skellingtons can lose limbs.

File: code/game/objects/items/stacks/sheets/sheet_types.dm [hunk 1]
```diff
@@ -161,3 +162,13 @@ var/list/datum/stack_recipe/charcoal_recipes = list ()
 /obj/item/stack/sheet/charcoal/New(var/loc, var/amount=null)
 		recipes = charcoal_recipes
 		return ..()
+
+		
+/obj/item/stack/sheet/bone
+	name = "bone"
+	desc = "Boney.  Probably has some marrow left"
```
Comment (ShiftyRail) [inline code/game/objects/items/stacks/sheets/sheet_types.dm]: You forgot a dot

Comment (Probe1) [reply code/game/objects/items/stacks/sheets/sheet_types.dm]: I thought the dot was included.

File: code/game/machinery/kitchen/gibber.dm [hunk 2]
```diff
@@ -207,8 +207,14 @@ obj/machinery/gibber/New()
 
 	var/obj/item/weapon/reagent_containers/food/snacks/meat/allmeat[totalslabs]
 	for (var/i=1 to totalslabs)
-		var/obj/item/weapon/reagent_containers/food/snacks/meat/newmeat = new occupant.meat_type(null, occupant)
-		newmeat.reagents.add_reagent (NUTRIMENT, sourcenutriment / totalslabs) // Thehehe. Fat guys go first
+		var/obj/item/weapon/newmeat		
+		if(istype(newmeat, /obj/item/weapon/reagent_containers))
```
Comment (Kurfursten) [inline code/game/machinery/kitchen/gibber.dm]: this check will always return false because at this point newmeat is null

Comment (Probe1) [reply code/game/machinery/kitchen/gibber.dm]: I thought I was free.

Comment (Kurfursten) [reply code/game/machinery/kitchen/gibber.dm]: the only species creating bones are ones without skin (plasmamen, skeletons) right?

just make this into a check like `!occupant.species.anatomy_flags & NO_SKIN`

Comment (Probe1) [reply code/game/machinery/kitchen/gibber.dm]: I'd like to leave it open for future expansion.

Comment (Kurfursten) [reply code/game/machinery/kitchen/gibber.dm]: In that case, just make the New() of your bone sheets compatible with receiving the occupant var and it won't matter

e.g.: do way with the if/else statement, just keep `newmeat = new occupant.meat_type(null, occupant)`

Add a `/obj/item/stack/sheet/bone/New(turf/loc, mob/occupant, var/amount)` that just calls parent (or you could change its name var to `[occupant]'s bones` etc. The important part is just to ensure that you don't get runtimes from handing an unexpected mob var to a sheet object

Comment (ghost) [reply code/game/machinery/kitchen/gibber.dm]: I considered that option when I was talking with Probe about his code @Kurfursten but personally I'm against deliberately feeding an object a parameter that it doesn't want, sanity or no sanity it just rubs me the wrong way and kinda feels like shit code

File: code/game/machinery/kitchen/gibber.dm [hunk 3]
```diff
@@ -207,8 +207,14 @@ obj/machinery/gibber/New()
...
-		var/obj/item/weapon/reagent_containers/food/snacks/meat/newmeat = new occupant.meat_type(null, occupant)
-		newmeat.reagents.add_reagent (NUTRIMENT, sourcenutriment / totalslabs) // Thehehe. Fat guys go first
+		var/obj/item/weapon/newmeat
+		if(istype(occupant.meat_type, /obj/item/weapon/reagent_containers))
+			newmeat = new occupant.meat_type(null, occupant)
+		else
+			newmeat = new occupant.meat_type()
+
+		if(istype(newmeat, /obj/item/weapon/reagent_containers))
+			newmeat.reagents.add_reagent (NUTRIMENT, sourcenutriment / totalslabs) // Thehehe. Fat guys go first
```
Comment (DamianX) [inline code/game/machinery/kitchen/gibber.dm]: move this after `newmeat = new occupant.meat_type(null, occupant)`, you're already typechecking there

**Review by DamianX — COMMENTED** [review]
Also, unlike meat, bones do not hold reagents and so the gibber won't transfer the dude's reagents to them. Intended? Feature?

File: code/game/machinery/kitchen/gibber.dm (hunk 3 above)
Comment (Probe1) [reply code/game/machinery/kitchen/gibber.dm]: Good call.

**Comment by Probe1** [comment]
I typed them as sheets for two reasons and I feel both made it the best choice.

1.  Functionality for future content.  If someone wants to make a bone derivative object it'll be hell to do that if it's a food that sanely inherits reagents from the mob holder.  

2.  What you eat doesn't really go into your bones anyway.

**Comment by DamianX** [comment]
But they're skellingtons, where else does the food go???

**Comment by Probe1** [comment]
Skellimemes can't eat food it just falls through them.

**Pull request merged**